modules/electrophysiology_browser/ajax/get_electrophysiology_session_data.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * Get EEG Session Data.
+ *
+ * This retrieves the entire eeg session data
+ * for a candidate to be visible on the eeg_session form.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  EEG_Session
+ * @author   Muhammad Khan <muhammad.khan@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+require_once 'BIDSFile.class.inc';
+
+/**
+ * User permission verification:
+ * Only users that have the eeg view all sites permission OR
+ * have the view site permission and belong to candidates session site
+ * are able to fetch the eeg session data.
+ */
+$user =& \User::singleton();
+$timePoint =& \TimePoint::singleton($_REQUEST['sessionID']);
+if (!$user->hasPermission('electrophysiology_browser_view_allsites')
+    && !((in_array($timePoint->getData('CenterID'), $user->getData('CenterIDs')))
+        && $user->hasPermission('electrophysiology_browser_view_site')
+    )
+)
+{
+    header('HTTP/1.1 403 Forbidden');
+    exit;
+}
+
+$response = getSessionData($_REQUEST['sessionID']);
+
+echo json_encode($response);
+
+// TODO Query code here to echo JSON data to the user.
+//echo json_encode(array('hello'=>'world'));
+// TODO Query code here to echo JSON data to the user.
+//echo json_encode(array('hello'=>'world'));
+function getSessionData($sessionID)
+{
+    $db = \Database::singleton();
+
+    $query = 'SELECT DISTINCT(pf.SessionID) FROM physiological_file pf LEFT
+    JOIN session s ON s.ID=pf.SessionID LEFT JOIN candidate c USING (CandID)
+    LEFT JOIN psc ON s.CenterID=psc.CenterID LEFT JOIN physiological_output_type
+    pot USING (PhysiologicalOutputTypeID) WHERE s.Active = "Y" AND pf.FileType
+    IN ("bdf", "cnt", "edf", "set", "vhdr", "vsm") ORDER BY pf.SessionID';
+
+    $sessions                = $db->pselect($query, array());
+    $sessions                = array_column($sessions, 'SessionID');
+    $response['patient']     = getSubjectData($sessionID);
+    $response['database']    = array_values(getFilesData($sessionID));
+    $response['sessions']    = $sessions;
+    $currentIndex            = array_search($sessionID,$sessions);
+    $response['nextSession'] = isset($sessions[$currentIndex+1]) ?
+        $sessions[$currentIndex+1] : '';
+    $response['prevSession'] = isset($sessions[$currentIndex-1]) ?
+        $sessions[$currentIndex-1] : '';
+
+    return $response;
+}
+function getSubjectData($sessionID)
+{
+    $subjectData = array();
+    $timePoint   =& \TimePoint::singleton($sessionID);
```

Comment:
```suggestion
    $timePoint   =& \TimePoint::singleton($sessionID);
```

Factory instead. I won't keeping on each place but it should be fixed in all cases.

Reply:
not sure I understand your comment. Could you explain it via Slack or the PR when you have some time? Thanks!